auth.go

```diff
@@ -61,9 +62,12 @@ type Auth struct {
 	banned         *set.Set
 	allowlist      *set.Set
 	ops            *set.Set
+	keynamesByFingerprint map[string]string
+	fingerprintsByKeyname map[string]string
```

Comment:
Why do we need `fingerprintsByKeyname`? I don't think we're actually using it?

Reply:
see Auth.Keyname() and Auth.KeynameFingerprint(). The former is used to determine what registereed name to assign a newly-connected user, based on their fingerprint. The latter is used to find the fingerprint that a nick may be registered to, to compare it with the fingerprint of the user's key in `/nick` and `/rename`. 

I suppose we could just build one map at launch, then implement one or the other function to iterate over the same map instead, instead of building two inverted maps to do separate lookups with.